sylt-compiler/src/typechecker.rs

```diff
@@ -726,6 +739,83 @@ impl TypeChecker {
     fn assignable(&mut self, assignable: &Assignable, ctx: TypeCtx) -> TypeResult<usize> {
         let span = assignable.span;
         match &assignable.kind {
+            AssignableKind::Variant { enum_ass, variant, value } => {
```

Comment:
This is almost the same as blobs. Maybe we can do something fancy when we have anonymous blobs.

Reply:
I don't think we can merge them in a good way. They're radically different... But you didn't really leave a mountain of evidence for your thoughts.

Reply:
I mean. The instantiation is pretty similar. Assuming we have anonymous blobs but also keep the old blob syntax, which one is an enum?
```
A { a: 1 }
A { a: 1 }
```

Reply:
Yeah, we'd need to go to anonymous blobs to make this decent. But at the moment there is no conflict, since an enum with a blob would have this syntax:
```
E.A B { a: 1 }
```
We'd have to remove the `A { a: 1 }` syntax - but that's when we add anonymous blobs.